### PR TITLE
Skilltrees/effect test

### PR DESCRIPTION
**Pull request by Joshuashou**
This pull request is to complete Sprint 4 issue of writing new tests for the skill trees that match implementation of our new effect features. We have successfully implemented such tests to be implemented into the effect_v2 branch of the chiventure project. This pull request will close that issue.

### CODE REVIEW

**Review by rohangupta2806 — APPROVED** [review]
Everything seems fine to me!  Approved.

File: tests/skilltrees/inventory_test.c [hunk 1]
```diff
@@ -113,135 +113,160 @@ Test(inventory_test, deep_copy_inventory_with_skills)
...
   int ret = inventory_has_skill(inventory, 1000, PASSIVE);
   cr_assert_eq(ret, -1,
     "Error: failed test inventory_has_skill_has_active\n");
 }
-*/
 
 /* Tests inventory_has_skill on active skill it does not have. */
-/*
+
 Test(inventory_tests, inventory_has_skill_has_not)
```
Comment (mariakravtsova) [inline tests/skilltrees/inventory_test.c:240]: The name of the test might be a little confusing - I recommend changing it to something like "inventory_does_not_have_skill"

File: tests/skilltrees/skilltree_test.c [hunk 2]
```diff
@@ -7,476 +7,607 @@
...
-//   int ret = inventory_skill_acquire(tree, inventory, skill2);
-
-//   cr_assert_eq(ret, 1,
-//     "Error: failed test inventory_skill_acquire_has on return value\n");
-//   cr_assert_eq(-1, inventory_has_skill(inventory, 1001, ACTIVE),
-//     "Error: failed test inventory_skill_acquire_has on adding to inventory\n");
-// }
+#include "test_init.h"
+
+/*Similar to skill_test.c, these tests must be rewritten to work with the new implementation of effects.  As the implementation nears completion, we will rewrite them.
```
Comment (mariakravtsova) [inline tests/skilltrees/skilltree_test.c:12]: This comment is no longer necessary? Seems like the tests mostly include the current implementation of effects are not commented out?

**Review by mariakravtsova — APPROVED** [review]
The tests seem to cover a lot of / all the possible cases for most functions, which is good. The pull request looks good to me!

File: tests/skilltrees/skilltree_test.c (hunk 2 above)
Comment (firatciftci) [reply tests/skilltrees/skilltree_test.c:12]: Yes, I think this comment should be removed at this point.

File: tests/skilltrees/skilltree_test.c [hunk 3]
```diff
@@ -7,476 +7,607 @@
...
+
+/* Tests skill_tree_free. */
+Test(skilltree_test, skill_tree_free_test)
+{
+    skill_tree_t* tree = skill_tree_new(1002, "this tree", 3);
+    int ret = skill_tree_free(tree);
+    cr_assert_eq(ret, SUCCESS, "Error: failed test skill_tree_free_test\n");
+}
+
+/*Tests after this line fail and are skipped.*/
```
Comment (firatciftci) [inline tests/skilltrees/skilltree_test.c:92]: Is this still the case?

File: tests/skilltrees/skilltree_test.c [hunk 4]
```diff
@@ -7,476 +7,607 @@
...
+    skill_tree_t* skilltree = skill_tree_new(1001, "this tree", 3);
+    skill_tree_node_add(skilltree, bomb_node);
+    skill_tree_node_add(skilltree, chop_node);
+    skill_tree_node_add(skilltree, inner_node);
+
+    skill_node_t** ret = get_all_skill_prereqs(skilltree, 1002);
+    cr_assert_null(ret,
+      "Error: failed test get_all_skill_prereqs_empty\n");
+    //cr_assert_eq(*out, 0,
+   //  "Error: failed test get_all_skill_prereqs_empty\n");
```
Comment (firatciftci) [inline tests/skilltrees/skilltree_test.c:422]: Make sure to remove any unused commented out code.

File: tests/skilltrees/skilltree_test.c [hunk 5]
```diff
@@ -7,476 +7,607 @@
...
+    inventory_skill_add(inventory, skill1);
+
+    int ret = inventory_skill_acquire(skilltree, inventory, skill2);
+
+    cr_assert_eq(ret, 0,
+      "Error: failed test inventory_skill_acquire_has on return value\n");
+    cr_assert_neq(-1, inventory_has_skill(inventory, 1001, ACTIVE),
+      "Error: failed test inventory_skill_acquire_has on adding to inventory\n");
+}
+*/
```
Comment (firatciftci) [inline tests/skilltrees/skilltree_test.c:613]: I think a single clarifying comment on why all of these tests are crashing and what should be done by students next year should be added just before this commented out code block.

**Review by firatciftci — COMMENTED** [review]
I have a few comments; once those are sorted out we can go ahead and merge the PR.

File: tests/skilltrees/inventory_test.c (hunk 1 above)
Comment (firatciftci) [reply tests/skilltrees/inventory_test.c:240]: +1 to @mariakravtsova's comment.

File: tests/skilltrees/skilltree_test.c (hunk 5 above)
Comment (rohangupta2806) [reply tests/skilltrees/skilltree_test.c:613]: @Joshuashou @m-ghosal I think this is happening due to the skill_has_prereqs method, so I would add a comment stating that that is causing a crash.

File: tests/skilltrees/skilltree_test.c (hunk 3 above)
Comment (Joshuashou) [reply tests/skilltrees/skilltree_test.c:92]: No it is not the case anymore, so I deleted the comment

File: tests/skilltrees/skilltree_test.c (hunk 4 above)
Comment (Joshuashou) [reply tests/skilltrees/skilltree_test.c:422]: Just did

File: tests/skilltrees/skilltree_test.c (hunk 5 above)
Comment (Joshuashou) [reply tests/skilltrees/skilltree_test.c:613]: Alright, added that comment on line 488 and wrote the other functions to refer to that comment

**Review by firatciftci — APPROVED** [review]
Everything looks good to me!

**Comment by Joshuashou** [comment]
Pull Request has successfully been merged. We have merged the successful changes in the testing features into the dev branch, including adding the sample-effects file that will streamline the system for writing tests for effects and skills easier, as well as edited the tests to better reflect changes in effect structures. This pull request fixed previous commented out tests that were written incorrectly. There is still work left to be done for some tests that currently crashed due to the get_all_prerequisite function not working properly, which will be in the backlog for next year's chiventure class.

**Comment by firatciftci** [comment]
Issue Score: ✔️++

Comments:
Great work!

**Pull request merged**